File: app/src/main/java/com/example/mike/mp3player/dagger/modules/MediaPlaybackServiceModule.java
package com.example.mike.mp3player.dagger.modules;

import com.example.mike.mp3player.dagger.components.MediaPlaybackServiceSubcomponent;
import com.example.mike.mp3player.service.MediaPlaybackService;

import dagger.Binds;
import dagger.Module;
import dagger.android.AndroidInjector;
import dagger.multibindings.ClassKey;
import dagger.multibindings.IntoMap;

@Module(subcomponents = MediaPlaybackServiceSubcomponent.class)
public abstract class MediaPlaybackServiceModule {

    @Binds
    @IntoMap
    @ClassKey(MediaPlaybackService.class)
    abstract AndroidInjector.Builder<?> provideMediaPlaybackService(
            MediaPlaybackServiceSubcomponent.Factory factory);

}

File: service/src/unitTest/java/com/github/goldy1992/mp3player/service/dagger/modules/MediaPlaybackServiceModule.kt
package com.github.goldy1992.mp3player.service.dagger.modules

import com.github.goldy1992.mp3player.service.MediaPlaybackService
import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ServiceComponent

@InstallIn(ServiceComponent::class)
@Module
abstract class MediaPlaybackServiceModule {

@Binds
abstract fun mediaPlaybackService(mediaPlaybackService: MediaPlaybackService) : MediaPlaybackService

}